.github/workflows/build-magma-linux.yml

```diff
@@ -0,0 +1,67 @@
+name: build-linux-magma
+
+on:
+  push:
+    branches:
+      main
+    paths:
+      - .ci/magma/*
+      - .ci/magma/package_files/*
+      - .github/workflows/build-magma-linux.yml
+  pull_request:
+    paths:
+      - .ci/magma/*
+      - .ci/magma/package_files/*
+      - .github/workflows/build-magma-linux.yml
+
+defaults:
+  run:
+    shell: bash -x -e -l {0}
+env:
+  BUILD_ENVIRONMENT: build-linux-magma
+  IN_CI: 1
+  IS_GHA: 1
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
+  cancel-in-progress: true
+
+jobs:
+  build-linux-magma:
+    if: github.repository_owner == 'pytorch'
+    runs-on: linux.2xlarge
+    strategy:
+      matrix:
+        cuda_version: ["124", "121", "118"]  # There is no pytorch/manylinux-cuda126 yet
+    steps:
+      - name: Checkout PyTorch
+        uses: actions/checkout@v4
+      - name: Build Magma Cuda
+        working-directory: .ci/magma
+        run: |
+          make magma-cuda${{ matrix.cuda_version }}
+      - name: Save as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: magma/output/linux-64/magma-cuda*.bz2
+      - name: Configure AWS credentials(PyTorch account)
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: arn:aws:iam::749337293305:role/gha_workflow_test_build_wheels
```

Comment:
I'm not quite sure about this role, I've seen there is a role for test and one for nightlies. 
Since this would be a new "channel" i.e. a new subfolder, it may need a new role